app/services/FilesSessionService.scala

```diff
@@ -17,38 +17,44 @@
 package services
 
 import config.ApplicationConfig
+import connectors.FilesSessionConnector
 import models.FileUploadStatus.{InProgress, NoFileSession, Ready, TimeExpiryError, UploadError}
-import models.{FileSession, FileUploadStatus}
+import models.{CreateFileSessionRequest, FileSession, FileUploadStatus}
 import org.joda.time.DateTime
 import play.api.Logging
-import repository.RasFilesSessionRepository
-import uk.gov.hmrc.mongo.cache.DataKey
+import play.api.libs.json.{JsError, JsSuccess, Json}
+import uk.gov.hmrc.http.HeaderCarrier
 
 import javax.inject.{Inject, Singleton}
 import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
-class RasFilesSessionService @Inject()(filesSessionRepository: RasFilesSessionRepository,
-                                       appConfig: ApplicationConfig)
-                                      (implicit ec: ExecutionContext) extends Logging {
+class FilesSessionService @Inject()(fileSessionConnector: FilesSessionConnector,
+                                    appConfig: ApplicationConfig) extends Logging {
 
   lazy val hoursToWaitForReUpload: Int = appConfig.hoursToWaitForReUpload
   private val STATUS_AVAILABLE: String = "AVAILABLE"
   private val defaultDownloadName: String = "Residency-status"
 
-  def createFileSession(userId: String, envelopeId: String): Future[Boolean] = {
-    filesSessionRepository.put[FileSession](userId)(DataKey("fileSession"), FileSession(None, None, userId, Some(DateTime.now().getMillis), None))
-      .map(_ => true)
-      .recover {
-      case ex: Throwable => logger.error(s"unable to create FileSession to cache => " +
-        s"$userId , envelopeId :$envelopeId,  Exception is ${ex.getMessage}")
-        false
+  def createFileSession(userId: String, envelopeId: String)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Boolean] = {
+    val createRequest = CreateFileSessionRequest(userId,envelopeId)
+    Json.toJson(createRequest).validate[CreateFileSessionRequest] match {
```

Comment:
Is it necessary to have the `.validate` etc. part here given that we just created the Json from the writes? The only way it would hit the `JsError` case would be if the reads and writes were incorrect which I feel could be tested at the unit level rather than checked here?